src/System.Management.Automation/engine/parser/Position.cs

```diff
@@ -145,9 +145,9 @@ internal static string VerboseMessage(IScriptExtent position)
             if (!string.IsNullOrEmpty(sourceLine))
             {
                 int spacesBeforeError = position.StartColumnNumber - 1;
-                int errorLength = (position.StartLineNumber == position.EndLineNumber)
+                int errorLength = (position.StartLineNumber == position.EndLineNumber && position.EndColumnNumber <= sourceLine.Length + 1)
                                       ? position.EndColumnNumber - position.StartColumnNumber
-                                      : sourceLine.TrimEnd().Length - position.StartColumnNumber + 1;
+                                      : sourceLine.Length - position.StartColumnNumber + 1;
```

Comment:
Do you know what was the original thought behind the addition of `TrimEnd`?

Reply:
Look above that, on line 142. TrimEnd() is called as sourceLine is set, and sourceLine isn't changed, so the additional call to TrimEnd() was redundant/unnecessary.